### PR TITLE
new interaction component for Blue/NanoS/NanoX

### PR DESCRIPTION
**Pull request by valpinkman**
Adds a new component to wrap all interactions of our devices (blue, nano s, nano x)
Adds a new step for blue device reset instruction before update
Replace illustration with new component where needed

### Type

Feature

### Context

LL-1372

### Parts of the app affected / Test plan

Manager -> Firmware Update
Send 
Receive 

---------------------------

Electron 2019-04-26 at 16 21 36

Electron 2019-04-26 at 16 21 53

[WIP] I still need to test the flow + fix the next slides with wording + illustrations

### CODE REVIEW

File: src/components/modals/UpdateFirmware/index.js [hunk 1]
```diff
@@ -46,27 +47,40 @@ const createSteps = ({ t }: { t: T }): Array<*> => {
...
 export type StepProps = DefaultStepProps & {
   firmware: FirmwareUpdateContext,
   onCloseModal: () => void,
   error: ?Error,
   setError: Error => void,
+  deviceModel: DeviceModel,
 }
 
-export type StepId = 'idCheck' | 'updateMCU' | 'finish'
+export type DeviceModel = 'blue' | 'nanoX' | 'nanoS'
```
Comment (gre) [inline src/components/modals/UpdateFirmware/index.js]: this type is already exported in `@ledgerhq/devices`

File: static/i18n/en/app.json [hunk 2]
```diff
@@ -308,7 +308,25 @@
         "Press the left button and hold it while you reconnect the USB cable until the <1><0>{{repairProcessing}}</0></1> screen appears",
       "mcuPin": "Please be patient while the update is finalized",
       "successTitle": "Firmware updated",
-      "successText": "Please re-install the apps on your device"
+      "successText": "Please re-install the apps on your device",
+      "resetTitle": "Prepare Device",
+      "resetSteps": {
+        "first": "1. Reset your device",
+        "connect": "Connect the Ledger Blue to your computer using the USB cable.",
```
Comment (gre) [inline static/i18n/en/app.json]: Ledger Blue should be a parameter

File: static/i18n/en/app.json [hunk 3]
```diff
@@ -308,7 +308,25 @@
...
+        "turnOn": "Press the button on the device to turn it on.",
+        "falsePin": "Enter three wrong PIN codes to reset the device.",
+        "turnOff": "Press and hold the side button until Power off device is displayed.",
+        "confirmTurnOff": "Tap Power off to confirm.",
+        "second": "2. Start In Bootloader Mode",
```
Comment (gre) [inline static/i18n/en/app.json]: Bootloader Mode should be a parameter (not to translate)

File: static/i18n/en/app.json [hunk 4]
```diff
@@ -308,7 +308,25 @@
...
-      "successText": "Please re-install the apps on your device"
+      "successText": "Please re-install the apps on your device",
+      "resetTitle": "Prepare Device",
+      "resetSteps": {
+        "first": "1. Reset your device",
+        "connect": "Connect the Ledger Blue to your computer using the USB cable.",
+        "turnOn": "Press the button on the device to turn it on.",
+        "falsePin": "Enter three wrong PIN codes to reset the device.",
+        "turnOff": "Press and hold the side button until Power off device is displayed.",
+        "confirmTurnOff": "Tap Power off to confirm.",
```
Comment (gre) [inline static/i18n/en/app.json]: If Power off is a wording on device, it should be a parameter

File: static/i18n/en/app.json [hunk 5]
```diff
@@ -308,7 +308,25 @@
...
+        "second": "2. Start In Bootloader Mode",
+        "boot": "Press and hold the side button until Boot Options is displayed",
+        "recoveryMode": "Tap Recovery Mode. Wait until the dashboard appears.",
```
Comment (gre) [inline static/i18n/en/app.json]: Recovery Mode is a parameter

File: static/i18n/en/app.json [hunk 6]
```diff
@@ -308,7 +308,25 @@
...
+        "third": "3. Uninstall all apps",
+        "openLive": "Open the Manager in Ledger Live.",
+        "uninstall":
+          "Click on the grey trash icon for all apps currently installed on the Ledger Blue. This makes room for the firmware installer.",
```
Comment (gre) [inline static/i18n/en/app.json]: Ledger Blue is a parameter

File: src/components/modals/UpdateFirmware/steps/01-step-install-full-firmware.js [hunk 7]
```diff
@@ -64,7 +65,6 @@ class StepFullFirmwareInstall extends PureComponent<Props, State> {
 
     if (!firmware.osu) {
       transitionTo('finish')
-      return
```
Comment (gre) [inline src/components/modals/UpdateFirmware/steps/01-step-install-full-firmware.js]: no longer required?

**Comment by MASAKO-NAGAI** [comment]
「BIT CLUB」仕様 の　Nano S　で　約　３年程前のです。クラッシュするので　アップデートして　Ledger Manager に接続してもダメ。操作方法教えて下さい。

File: src/components/FlashMCU.js [hunk 8]
```diff
@@ -21,18 +23,21 @@ const Separator = styled(Box).attrs({
...
-        {t('manager.modal.mcuFirst')}
+        <Trans i18nKey="manager.modal.mcuFirst" />
       </Text>
-      <img
-        src={i('logos/unplugDevice.png')}
-        style={{ width: '100%', maxWidth: 368, marginTop: 30 }}
-        alt={t('manager.modal.mcuFirst')}
-      />
+      <Box mt={5}>
+        {/* TODO: add connecting / disconnecting to NanoX */}
```
[the remark this answers is not in the document]
Comment (valpinkman) [reply src/components/FlashMCU.js]: yes

**Pull request merged**